File: src/main/java/komu/blunt/core/CoreEmptyExpression.java
package komu.blunt.core;

import komu.blunt.asm.Assembler;
import komu.blunt.asm.Instructions;
import komu.blunt.asm.Linkage;
import komu.blunt.asm.Register;

public final class CoreEmptyExpression extends CoreExpression {

    public static final CoreExpression INSTANCE = new CoreEmptyExpression();

    private CoreEmptyExpression() { }
    
    @Override
    public Instructions assemble(Assembler asm, Register target, Linkage linkage) {
        Instructions instructions = new Instructions();
        instructions.finishWithLinkage(linkage);
        return instructions;
    }
}

File: src/main/kotlin/komu/blunt/core/CoreEmptyExpression.kt
package komu.blunt.core

import komu.blunt.asm.*

class CoreEmptyExpression private () : CoreExpression() {

    class object {
        val INSTANCE = CoreEmptyExpression()
    }

    override fun assemble(asm: Assembler, target: Register, linkage: Linkage) =
        instructions {
            finishWithLinkage(linkage)
        }

    override fun simplify() = this

    override fun toString() = "#empty"
}
